asserts/assertstest/assertstest.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package assertstest provides helpers for testing code that involves assertions.
+package assertstest
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os/exec"
+	"time"
+
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/packet"
+
+	"github.com/snapcore/snapd/asserts"
+)
+
+// GenerateKey generates a private/public key pair of the given bits. It panics on error.
+func GenerateKey(bits int) (asserts.PrivateKey, *packet.PrivateKey) {
+	priv, err := rsa.GenerateKey(rand.Reader, bits)
+	if err != nil {
+		panic(fmt.Errorf("failed to create private key: %v", err))
+	}
+	pkt := packet.NewRSAPrivateKey(time.Now(), priv)
+	return asserts.OpenPGPPrivateKey(pkt), pkt
+}
+
+// ReadPrivKey reads a PGP private key (either armored or simply base64 encoded). It panics on error.
+func ReadPrivKey(pk string) (asserts.PrivateKey, *packet.PrivateKey) {
+	rd := bytes.NewReader([]byte(pk))
+	blk, err := armor.Decode(rd)
+	var body io.Reader
+	if err == nil {
+		body = blk.Body
+	} else {
+		rd.Seek(0, 0)
+		// try unarmored
+		body = base64.NewDecoder(base64.StdEncoding, rd)
+	}
+	pkt, err := packet.Read(body)
+	if err != nil {
+		panic(err)
+	}
+
+	pkPkt := pkt.(*packet.PrivateKey)
+	return asserts.OpenPGPPrivateKey(pkPkt), pkPkt
+}
+
+// A sample developer key.
```

Comment:
Why *developer* key? Surely this could be used for other reasons too, e.g. as a root key.